app/javascript/lib/validation/validate-file.js

```diff
@@ -0,0 +1,121 @@
+/**
+* @fileoverview Validates Single Cell Portal files on the user's computer
+*/
+
+import { readLinesAndType } from './io'
+
+/** Remove white spaces and quotes from a string value */
+function clean(value) {
+  return value.trim().replaceAll(/"/g, '')
+}
+
+/**
+ * Verify headers are unique and not empty
+ */
+function validateUniqueHeaders(headers) {
+  const issues = []
+  const uniques = new Set(headers)
+
+  // Are headers unique?
+  if (uniques.size !== headers.length) {
+    const seen = new Set()
+    const duplicates = new Set()
+    headers.forEach(header => {
+      if (header in seen) {duplicates.add(header)}
+      seen.add(header)
+    })
+
+    const dupString = [...duplicates].join(', ')
+    const msg = `Duplicate header names are not allowed: ${dupString}`
+    issues.push(['error', 'format', msg])
+  }
+
+  // Are all headers non-empty?
+  if (uniques.has('')) {
+    const msg = 'Headers cannot contain empty values'
+    issues.push(['error', 'format', msg])
+  }
+
+  return issues
+}
+
+/**
+ * Verify second row starts with TYPE (case-insensitive)
+ */
+function validateTypeKeyword(annotTypes) {
+  const issues = []
+
+  const value = annotTypes[0]
+
+  if (value.toUpperCase() === 'TYPE') {
+    if (value !== 'TYPE') {
```

Comment:
What is this trying to check? The comment made me think "type" or "TYPE" would be acceptable but this check makes it seem like only "TYPE" is ok?

Reply:
Great question.  

Like the [reference implementation](https://github.com/broadinstitute/scp-ingest-pipeline/blob/development/ingest/annotations.py#L258) in Ingest Pipeline, if the provided header is "type", then it passes a case-insensitive match but fails a case-sensitive match and thus gets a warning (see line 53 here).  

However, if the provided header is e.g. "foo", then it doesn't even pass a case-insensitive match, and thus gets an error.

If the provided header is "TYPE", then it passes a case sensitive match, and gets neither a warning nor an error.